onmt/Models.py

```diff
@@ -4,59 +4,80 @@
 from torch.autograd import Variable
 import onmt
 import onmt.modules
+from onmt.IO import ONMTDataset
 from onmt.modules import aeq
 from onmt.modules.Gate import ContextGateFactory
 from torch.nn.utils.rnn import pad_packed_sequence as unpack
 from torch.nn.utils.rnn import pack_padded_sequence as pack
 
 
 class Embeddings(nn.Module):
-    def __init__(self, vec_size, opt, dicts, feature_dicts=None):
+    """
+    Words embeddings dictionary for Encoder/Decoder.
+
+    Args:
+        embedding_dim (int): size of the dictionary of embeddings.
+        position_encoding (bool): use a sin to mark relative words positions.
+        feat_merge (string): merge action for the features embeddings:
+                    concat, sum or mlp.
+        feat_dim_exponent (float): when using '-feat_merge concat', feature
+                    embedding size is N^feat_dim_exponent, where N is the
+                    number of values of feature takes.
+        feat_embedding_dim (int): embedding dimension for features when using
+                    '-feat_merge mlp'
+        dropout (float): dropout probablity.
+        dicts (Vocab): vocabulary dictionary.
```

Comment:
Instead of passing a `dict` can you instead just pass its length and call it `num_word_embeddings` similary for the `features_dicts` . Whenever possible let's not pass `Vocab` objects around.